src/server/rdb_save.cc

```diff
@@ -337,6 +337,13 @@ error_code RdbSerializer::SelectDb(uint32_t dbid) {
 // Called by snapshot
 io::Result<uint8_t> RdbSerializer::SaveEntry(const PrimeKey& pk, const PrimeValue& pv,
                                              uint64_t expire_ms, DbIndex dbid) {
+  string_view key = pk.GetSlice(&tmp_str_);
```

Comment:
Looks like we only need this inside the `if`, so maybe insert it there?
This could save an unneeded allocation (that's rarely needed for the sad path)

Reply:
it's also used later in line 365 but I agree